packages/@ourworldindata/grapher/src/controls/SettingsMenu.tsx

```diff
@@ -0,0 +1,456 @@
+        if (!availableFacetStrategies.includes(FacetStrategy.none)) return true
+
+        const showFacetControlChartType = [
+            StackedArea,
+            StackedBar,
+            StackedDiscreteBar,
+            LineChart,
+        ].includes(type)
+
+        const hasProjection = filledDimensions.some(
+            (dim) => dim.display.isProjection
+        )
+
+        return (
+            showFacetControlChartType &&
+            !hideFacetControl &&
+            !hasProjection &&
+            !isOnTableTab
+        )
+    }
+
+    @computed get showTableFilterToggle(): boolean {
+        const { hideTableFilterToggle, selection } = this.manager
+        const hasSelection =
+            selection instanceof SelectionArray
+                ? selection.hasSelection
+                : (selection?.length ?? 0) > 0
+
+        return hasSelection && !hideTableFilterToggle
+    }
+
+    @computed get showSettingsMenuToggle(): boolean {
+        if (this.manager.isOnMapTab) return false
+        if (this.manager.isOnTableTab) return this.showTableFilterToggle
+
+        return !!(
+            this.showYScaleToggle ||
+            this.showXScaleToggle ||
+            this.showFacetYDomainToggle ||
+            this.showZoomToggle ||
+            this.showNoDataAreaToggle ||
+            this.showFacetControl ||
+            this.showAbsRelToggle
+        )
+
+        // TODO: add a showCompareEndPointsOnlyTogggle to complement compareEndPointsOnly
+    }
+
+    componentDidMount(): void {
+        document.addEventListener("click", this.onDocumentClick, {
+            capture: true,
+        })
+    }
+
+    componentWillUnmount(): void {
+        document.removeEventListener("click", this.onDocumentClick)
+    }
+
+    @action.bound onDocumentClick(e: MouseEvent): void {
+        if (
+            this.active &&
+            this.contentRef?.current &&
+            !this.contentRef.current.contains(e.target as Node) &&
+            document.contains(e.target as Node)
+        )
+            this.toggleVisibility()
+    }
+
+    @action.bound toggleVisibility(e?: React.MouseEvent): void {
+        this.active = !this.active
+        if (this.active) this.visible = true
+        this.drawer?.classList.toggle("active", this.active)
+        e?.stopPropagation()
+    }
+
+    @action.bound onAnimationEnd(): void {
+        if (!this.active) this.visible = false
+    }
+
+    @computed get manager(): SettingsMenuManager {
+        return this.props.manager
+    }
+
+    @computed get chartType(): string {
+        const { type } = this.manager
+        return type.replace(/([A-Z])/g, " $1")
+    }
+
+    @computed get drawer(): Element | null {
+        // use a drop-down menu when embedded in an iframe
+        if (this.manager.isInIFrame) return null
+        // use the drawer `<nav>` element if it exists, otherwise render into a drop-down menu
+        return this.manager.base?.current?.closest(".related-charts")
```

Comment:
I put the `isInIFrame` check in after testing what Grapher looks like when embedded via an iFrame on foreign sites. It needs the additional check because what's embedded is `ourworldindata.org/grapher/slug`, so it does find a `nav#grapher-settings-id` on the page, but a drop-down is more sensible for iframe embedded charts, I think.

Reply:
Okay okay, I understand.
There might be a better way to detect a data page though?

Otherwise, could you please write this up as a code comment so this knowledge is not lost in the future?

Reply:
If I'm not mistaken, `grapher.isEmbeddedInAnOwidPage` is true for all embedded charts, including the related charts blocks on data pages. (Would be good to double-check though, I'm not 100% sure)

Reply:
I think we must have had dueling edits at one point too, since I don't see any mention of `isInIFrame` in the branch's current version of [`SettingsMenu.tsx`](https://github.com/owid/owid-grapher/blob/grapher-redesign/packages/%40ourworldindata/grapher/src/controls/SettingsMenu.tsx)...

Reply:
that's weird.. I'm not sure how that would happen. It seems like it got lost in [this commit](https://github.com/owid/owid-grapher/pull/2732/commits/053fe9aa7978e61373ec8eba9ddd4df834dd24e1)

Reply:
I'm actually very confused about this. To me it looks like you deleted it?

Reply:
Yeah it's a mystery to me as well. I wonder if I had unsaved changes in my editor before pulling and then overwrote those lines when I saved? In any case, thanks for the link so I can add it back.

Reply:
I'm now letting either `isEmbeddedInAnOwidPage` or `isInFrame` put the settings menu into drop-down mode. It seems to be working as intended on data pages: the main graphic uses the drawer and the Related Charts instances use the drop-down.

Reply:
Nice!